wp/plugin/lib/equality.ml

```diff
@@ -0,0 +1,387 @@
+open !Core_kernel
+open Bap.Std
+open Graphlib.Std
+open Base__Option.Monad_infix
+
+(* some of these pattern matchings are massive; we don't want fragile warnings *)
+[@@@warning "-4"]
+
+module H = Bap.Std.Graphs.Ir
+module J = Graphlib.To_ocamlgraph (H)
+module BFS = Graph.Traverse.Bfs (J)
+module TidSet = Tid.Set
+
+module TidMap = Tid.Map
+module VarMap = Var.Map
+module VarSet = Var.Set
+module TidTupleMap = Map.Make(struct type t = Tid.t * Tid.t [@@deriving sexp, compare] end)
+
+type varToVarMap = Var.t VarMap.t
```

Comment:
I would prefer you not define this type synonym. `Var.t VarMap.t` is clearer and more self explanatory than `varToVarMap`

Reply:
(also, `var` is an alias to `Var.t` if that helps)